--- patchutils/filterdiff.c:1.27	Sat Mar  9 20:34:17 2002
+++ patchutils/filterdiff.c	Wed Mar 20 10:21:49 2002
@@ -176,12 +176,14 @@
 	 *   from lines... (omitted if there are only insertions)
 	 * --- start[,end] ----
 	 *   to lines... (omitted if there are only deletions)
+	 *[*** start[,end] ****
+	 * ...]
 	 */
 
-	for (;;) {
-		if (strcmp (*line, "***************\n"))
-			return 1;
+	if (strcmp (*line, "***************\n"))
+		return 1;
 
+	for (;;) {
 		to_omitted = 1; // until we know that to lines may not be
 
 		for (i = 0; i < 2; i++) {
@@ -194,6 +196,14 @@
 			if (match && mode == mode_filter)
 				fputs (*line, stdout);
 
+			if (!i && !strcmp (*line, "***************\n")) {
+				/* Some diffs seem to have this for every
+				 * set of changes.  SuSV2 says not to,
+				 * but the GNU diff info page disagrees. */
+				i--;
+				continue;
+			}
+
 			if (strncmp (*line, i ? "--- " : "*** ", 4))
 				return 1;
 
@@ -259,9 +269,6 @@
 					to_omitted = 0;
 			}
 		}
-
-		if (getline (line, linelen, f) == -1)
-			return EOF;
 	}
 
 	return 0;
