diff -urN patchutils-0.2.11.org/Makefile.am patchutils-0.2.11/Makefile.am
--- patchutils-0.2.11.org/Makefile.am	Fri Apr 19 21:01:05 2002
+++ patchutils-0.2.11/Makefile.am	Fri Apr 19 21:01:44 2002
@@ -26,14 +26,14 @@
 	ln -sf $< $@
 
 install-exec-hook:
-	ln -sf interdiff$(EXEEXT) $(bindir)/combinediff$(EXEEXT)
-	ln -sf filterdiff$(EXEEXT) $(bindir)/lsdiff$(EXEEXT)
-	ln -sf filterdiff$(EXEEXT) $(bindir)/grepdiff$(EXEEXT)
+	ln -sf interdiff$(EXEEXT) $(DESTDIR)$(bindir)/combinediff$(EXEEXT)
+	ln -sf filterdiff$(EXEEXT) $(DESTDIR)$(bindir)/lsdiff$(EXEEXT)
+	ln -sf filterdiff$(EXEEXT) $(DESTDIR)$(bindir)/grepdiff$(EXEEXT)
 
 uninstall-hook:
-	rm $(bindir)/combinediff$(EXEEXT)
-	rm $(bindir)/lsdiff$(EXEEXT)
-	rm $(bindir)/grepEXEEXT)
+	rm $(DESTDIR)$(bindir)/combinediff$(EXEEXT)
+	rm $(DESTDIR)$(bindir)/lsdiff$(EXEEXT)
+	rm $(DESTDIR)$(bindir)/grepEXEEXT)
 
 CLEANFILES=combinediff lsdiff grepdiff
 MAINTAINERCLEANFILES=$(man_MANS)
